File: 프로그래머스/lv2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    // 각 기능별 배포 가능일
    const days = progresses.map((p, i) => Math.ceil((100 - p)/speeds[i])); 
    const pushDay = [];
    
    let start = days[0];
    let utils = 1;
    
    for(let i = 1; i < days.length; i++) {
        // 시작보다 큰 일정 발견
        if(start < days[i]) {
            pushDay.push(utils);
            
            start = days[i];
            utils = 1;
        }
        else {
            utils++;
        }
    }
    
    pushDay.push(utils);

    return pushDay;
}

/*
일단 각 기능별 배포일을 배열에 담기 
배포일이 시작보다 클 때까지 쭉 읽는다.
시작보다 큰 일정을 만나면 새로운 날에 배포하는 게 된다. 

5, (10, 1, 1), (20, 1)
*/

File: 프로그래머스/2/42586. 기능개발/기능개발.py
import math

def solution(progresses, speeds): 
    cursor = 0
    ans = []
    days = 0
    while(True):
        if cursor >= len(progresses):
            break
        if cursor + 1 == len(progresses):
            ans.append(1)
            break
            
        cur_progress = progresses[cursor] + (speeds[cursor] * days)
        
        days += math.ceil(((100 - cur_progress) / speeds[cursor]))
        
        c = 1
        
        start = cursor + 1
        for i in range(start, len(progresses)):
            cp = progresses[i]
            if cp + (speeds[i] * days) >= 100:
                c += 1
                cursor = i
            else:
                break
                
        cursor += 1
        ans.append(c)
        
    return ans
                
                
                